### PR TITLE
Update Contributing Page

### PR DESCRIPTION
**Pull request by alexfornuto**
- Replaces `--recursive` (which is unnecessary and slows down the process) with `--remote` (which is required to avoid build errors on first run).
- Adds a step to switch to `master` or a versioned branch before creating a new one. The submodule inits detached at the latest commit.

Supersedes #15741, which I'm replacing with this branch because it follows naming convention and is not forked (and I killed my local copy setting up the dev environment).

### CODE REVIEW

File: docs/pages/contributing/documentation/how-to-contribute.mdx
Comment (alexfornuto) [reply docs/pages/contributing/documentation/how-to-contribute.mdx:33]: My only concern with this change comes from the [documentation](https://git-scm.com/docs/git-switch) for `switch`:
>THIS COMMAND IS EXPERIMENTAL. THE BEHAVIOR MAY CHANGE

File: docs/pages/contributing/documentation/how-to-contribute.mdx
Comment (alexfornuto) [reply docs/pages/contributing/documentation/how-to-contribute.mdx:21]: I'll have to do some more testing to confirm, but when I first set up my local docs dev environment, the local server wouldn't build until I ran this command with `--remote`.

Comment (alexfornuto) [reply docs/pages/contributing/documentation/how-to-contribute.mdx:21]: [This gist](https://gist.github.com/alexfornuto/9debd1aff9aff16031cdbcf6e61d280c) demonstrates the issue, and it's resolution with `--remote`.

Comment (zmb3) [reply docs/pages/contributing/documentation/how-to-contribute.mdx:22]: Hmm. I have no idea. @avatus does this look alright to you?

Comment (avatus) [reply docs/pages/contributing/documentation/how-to-contribute.mdx:22]: Remote is correct here but it's better to point to the yarn script we already have for this `yarn git-update` which runs `git submodule update --remote`.

Two different scenarios when people would want to update. 
1. They just cloned the repo
2. they want to pull changes to the submodules.
In the `README` we separate this into two steps. 
```
git submodule init
```
and then
```
yarn git-update
```

According [to the docs](https://www.linux.org/docs/man1/git-submodule.html), we can forgo the initial `git submodule init` and just use `--init` if we plan on not updating any submodule locations (we don't. thats a separate thing when we add versions).

IMO, rather than adding `git submodule update --init --remote`, we add the `--init` flag to our `yarn git-update` script and specify that here instead.

Comment (alexfornuto) [reply docs/pages/contributing/documentation/how-to-contribute.mdx:22]: >IMO, rather than adding git submodule update --init --remote, we add the --init flag to our yarn git-update script and specify that here instead.
That was my [first instinct](https://gravitational.slack.com/archives/CM7E1200L/p1661202855505829?thread_ts=1661202667.452759&cid=CM7E1200L), but I decided against it because I wasn't sure if adding `--init` once submodules were already initialized would be a problem.

Comment (avatus) [reply docs/pages/contributing/documentation/how-to-contribute.mdx:22]: I think adding it will be fine.
from the docs: 

--init
This option is only valid for the update command. Initialize all submodules for which "git submodule init" **has not been called so far** before updating.

Seems like docs are saying, and from my testing, it doesn't do anything if they've already been init so its idempotent in that sense.

Comment (alexfornuto) [reply docs/pages/contributing/documentation/how-to-contribute.mdx]: I've updated this branch to use the yarn script. This PR is now blocked until the docs repo is updated to include `--init` in the script. I just discovered that I don't have push rights to the docs branch, so I'm waiting on that to push my fixing branch, which this PR will now be dependent on.

**Review by ptgott — APPROVED** [review]

**Review by avatus — APPROVED** [review]

**Review by zmb3 — APPROVED** [review]

**Pull request merged**
